### ISSUE TITLE
Suggestion: resetting a variable (e.g. "var") shouldn't automatically reset "var+" and "var-"

### ISSUE THREAD

**Issue by aeronaut999** [opentx/opentx#8869]
Suggestion-- when any given variable such as "Var" is "Reset", the "+" and "-" versions of that variable (e.g. "Var+" and "Var-")  should _not_ automatically be reset.

This would avoid the need to write special programming lines to remember the maximum (or minimum) value of a variable that may be reset several times during a flight.

If there is a special reason to reset the "+" and/or "-" versions of the variable, that can always be accomplished with one or two extra programming lines in the "Special Functions" area.

**Comment by bigtwin** [comment]
I am curios and its not really clear to me what this is about ...
What value would I want to reset if not a maximum of e.g. _Alt_ or the minimum of e.g. _VFAS_? I am doing this rather often using "_Reset Telemetry_". 
How is this related to your suggestion?

**Comment by aeronaut999** [comment]
bigtwin here is a specific example.  I run three altimeter variables on my motorgliders.  One gets reset at the start of the flying session, i.e. when the motor kill switch is first deactivated with that specific model selected.  One gets continually reset whenever the motor is running, so I can keep track of my *power-off* altitude gain.  And one gets continually reset whenever the motor is running *or* whenever the value of that altimeter drops below zero, so I can keep track of the altitude gained above the point where the motor was last cut *or* above the lowest power-off "low point" following the motor cut, whichever of these two gives the bigger altitude number.  (Example: powered climb to 800', cut motor, descend to 400', find thermal, power-off climb to 1400', this altimeter will now read 1000', because it was last reset when the glider was 400' over the ground.)  Call this Alt3.  I would like for Alt3+ not to get reset whenever Alt3 is reset.  So that if later in the same flight, I sink out to 100', and then find a thermal and climb to 200', at which point Alt3 will read 100' (because it was last reset when the glider was 100' above the ground), Alt3+ will still read 1000'.   This could be accomplished via the suggestion above.

**Comment by bigtwin** [comment]
Wow! Thx for the example/explanation. I never heard of something like that. And actually I dont not see a need for such a sophisticated config on my planes.

**Comment by kilrah** [comment]
You can duplicate Alt3 to a new sensor that you don't reset, it'll then keep the +/- and you just don't look at the value itself.

**Comment by aeronaut999** [comment]
Kilrah, I don't see any way to duplicate a sensor in a way that would cause it to remember the value 1000' in the example given above.  Do you?  If so please explain more, because I am missing something.  

Of course it is simple to duplicate a sensor that would end up remembering the value 1400' in the example given above.  Simply make another altimeter that is an exact duplicate of the first, and never reset it.  I've already done that.  

OK, is this what you mean--
New sensor (call it "alt4") is a "calculated" sensor
Formula is "add"
Source is "alt3" 
Then no matter how many times I reset "alt3", "alt4+" will be the maximum value that alt 3 ever reached during that flight or flying session..

Would that work?  Is the program ok with having only one source for an "add" calculation?

**Comment by aeronaut999** [comment]
Re -- 

"New sensor (call it "alt4") is a "calculated" sensor
Formula is "add"
Source is "alt3" 
Then no matter how many times I reset "alt3", "alt4+" will be the maximum value that alt 3 ever reached during that flight or flying session.

Would that work?  Is the program ok with having only one source for an "add" calculation?"

Works perfectly, mission accomplished.  

(Actually I used formula "max" rather than "add", but "add" would have surely given the same result.  The point is simply to make "alt4" be _always_ equal to alt3.  Whenever alt 3 gets "reset", alt4 also goes to zero, but since no "reset" command is directly applied to alt4, the value of alt4+ is unaffected by the "reset" command.)

Suggestion withdrawn --

Issue https://github.com/opentx/opentx/issues/7949 is also now of somewhat less _practical_ interest to me,  _but still describes an interesting flaw in the logic of OpenTX_.

**Comment by kilrah** [comment]
Cool.

> (Actually I used formula "max" rather than "add", but "add" would have surely given the same result. The point is simply to make "alt4" be always equal to alt3.

Simply duplicatting Alt3 would have achieved the same result, hence being what I initially suggested.

Given the complexity of that setup it seems more appropriate for a lua script though, where you could make a custom display that shows the things you care about in a more convenient way, don't know if it's the case for your particular use but there are people who have made that kind of thing for glider stats long ago already.

**Comment by aeronaut999** [comment]
> Cool.
> 
> > (Actually I used formula "max" rather than "add", but "add" would have surely given the same result. The point is simply to make "alt4" be always equal to alt3.
> 
> Simply duplicatting Alt3 would have achieved the same result, hence being what I initially suggested.

I don't think so, unless I'm misunderstanding what you mean by "duplicate".  Is there a specific "duplicate" command, or do you just mean making a copy of Alt3 in the sensor editing page?

See, the thing is that for things to work as I want, the Alt3 sensor *does* have to be reset to zero multiple times during the flight.  To make Alt4 work as I want, I need a way to set Alt4 to zero too, but without actually applying a reset command, which would wipe out Alt4+.  

If you understand what I'm trying to do here and see an alternate way to accomplish it, I'd be curious to understand more, but I'm not seeing it.  

I already have a different altimeter that is zeroed at launch and never reset during the flight, that's not what Alt3 is supposed to do.  It's supposed to keep track of the climb from the lowest low point after the end of the latest motor run.  It automatically resets to zero whenever the motor is running or whenever it goes negative.   So this resetting may happen multiple times during a flight.  And I want to remember the highest value ever reached by Alt3 in a given flight.  If I don't run Alt4 as a "calculated" sensor from Alt3, then the only options left as far as I can see is (a) resetting Alt4 too, which resets Alt4+, which I don't want, or (b) not resetting Alt4, in which case it's just an altimeter measuring model's total height above ground, which I already have.

Anyway, more than enough said, unless you think I'm missing something in which case I'd be curious to hear more--

Thanks

**Comment by kilrah** [comment]
OK, I guess you're using the auto offset, which I missed.

**Comment by aeronaut999** [comment]
> OK, I guess you're using the auto offset, which I missed.

I think there's more to it than that, it has to do with intention to reset the variables whenever the values go negative.   To keep track of the climb from the lowest low point following the end of the latest motor run.

**Comment by 3djc** [comment]
Really this scenario would be sooo much better suited for LUA, you could really make some thing very nice using that

**Comment by aeronaut999** [comment]
> Really this scenario would be sooo much better suited for LUA, you could really make some thing very nice using that

It is working perfectly fine this way too-- I have yet to dive into writing and using LUA scripts-- and am kind of afraid to start because I have a whole bunch of ideas for other little projects I'd want to do, would take up all my flying time!

Anyway, suggestion withdrawn--

**Comment by jrwieland** [comment]
> > Really this scenario would be sooo much better suited for LUA, you could really make some thing very nice using that
> 
> It is working perfectly fine this way too-- I have yet to dive into writing and using LUA scripts-- and am kind of afraid to start because I have a whole bunch of ideas for other little projects I'd want to do, would take up all my flying time!
> 
> Anyway, suggestion withdrawn--

I have free time and dumb enough to offer to write a script, based on your input, to get you the information on the screen that your after.  With a script you can  have a chart showing each of your ascends plus any other items that you would desire.  Literally,  it can be designed to put anything on the screen.  here is a screenshot of the one that I have for my helicopters (food for thought).  Just let me know if your interested
![clrfeat](https://user-images.githubusercontent.com/20526716/156011429-f6dac484-be93-4bb7-b633-e589b0c4482c.png)
